File: solutions/clojure/binary-search/2/src/binary_search.clj
(ns binary-search)

(defn search-for [key numbers]
    (loop [low 0 high (dec (count numbers))] 
        (if (> low high) -1
        (let [middle (quot (+ low high) 2)
              value (nth numbers middle)]
            (cond 
                (< value key) (recur (inc middle) high)
                (> value key) (recur low (dec middle))
                (== value key) middle
            )
        ))
    )
)

File: solutions/python/binary-search/2/binary_search.py
def find(list, value):
    high = len(list)-1
    low = 0
    while low<=high:
        mid = (high + low)//2
        if list[mid]==value:
            return mid
        low, high =  (mid+1, high) if list[mid]<value else (low, mid-1)
    raise ValueError("value not in array")
        
    
